gtsfm/utils/geometry_comparisons.py

```diff
@@ -40,6 +40,52 @@ def align_rotations(aRi_list: List[Rot3], bRi_list: List[Rot3]) -> List[Rot3]:
     return [aRb.compose(bRi) for bRi in bRi_list]
 
 
+def align_poses_sim3_wrapper(aTi_list: List[Optional[Pose3]], bTi_list: List[Optional[Pose3]]) -> List[Optional[Pose3]]:
+    """Align by similarity transformation, but allow missing estimated poses in the input.
+
+    Note: this is a wrapper for align_poses_sim3() that allows for missing poses/dropped cameras.
+    This is necessary, as align_poses_sim3() requires a valid pose for every input pair.
+
+    We force SIM(3) alignment rather than SE(3) alignment.
+    We assume the two trajectories are of the exact same length.
+
+    Args:
+        aTi_list: reference poses in frame "a" which are the targets for alignment
+        bTi_list: input poses which need to be aligned to frame "a"
+
+    Returns:
+        aTi_list_: transformed input poses previously "bTi_list" but now which
+            have the same origin and scale as reference (now living in "a" frame)
+    """
+    assert len(aTi_list) == len(bTi_list)
+
+    # only choose target poses for which there is a corresponding estimated pose
+    corresponding_aTi_list = []
+    valid_camera_idxs = []
+    dropped_camera_idxs = []
+    valid_bTi_list = []
+    for i, bTi in enumerate(bTi_list):
+        if bTi is None:
+            dropped_camera_idxs.append(i)
+        else:
+            valid_camera_idxs.append(i)
+            valid_bTi_list.append(bTi)
+            corresponding_aTi_list.append(aTi_list[i])
+
+    if len(dropped_camera_idxs) > 0:
```

Comment:
I think this should be logged in whichever module is dropping these (if necessary), unnecessary here.

Reply:
I agree that logging here is not very useful. But I don't think we need to log it anywhere. We dont care about the exact indices anyway. We can just log (and I think we are already doing it) the number of valid cameras.

Reply:
sure, i'll remove it